addons/sale_coupon/models/sale_order.py

```diff
@@ -157,7 +157,11 @@ def _get_reward_values_discount(self, program):
             }]
         reward_dict = {}
         lines = self._get_paid_order_lines()
-        amount_total = sum(self._get_base_order_lines(program).mapped('price_subtotal'))
+        base_lines = self._get_base_order_lines(program)
+        if any(base_lines.mapped('tax_id.price_include')):
+            amount_total = sum(base_lines.mapped('price_total'))
+        else:
+            amount_total = sum(base_lines.mapped('price_subtotal'))
```

Comment:
This seems quite wrong reading it.
Shouldn't this be a case by case depending on the line's tax?
Eg, adding the total when price_include and subtotal if not, but do that for each line to add in amount_total.

If you have 2 lines, one with price_included and one with price_excluded, we won't have the correct amount_total?
(not even talking about multiple taxes cases :grimacing: )